api/app/src/main/kotlin/packit/service/OrderlyRunnerClient.kt

```diff
@@ -0,0 +1,63 @@
+package packit.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.http.*
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestTemplate
+import packit.AppConfig
+import packit.exceptions.PackitException
+import packit.model.ServerResponse
+import packit.model.dto.OrderlyRunnerVersion
+
+interface OrderlyRunner
+{
+    fun getVersion(): OrderlyRunnerVersion
+}
+
+@Service
+class OrderlyRunnerClient(appConfig: AppConfig) : OrderlyRunner
```

Comment:
There's a lot of shared code between this client and the outpack client isn't there? in terms of Anmol's comment about error handling etc, and dealing with standard Porcelain Server responses (which we should be getting from both Outpack and Orderly), it feels like that could be delegated to a generic client class, and then the orderly client and outpack clients could provide methods for their various offerings (e.g. `getVersion`) which would invoke the correct endpoint via the generic client.

Reply:
@absternator